### PR TITLE
Adding notebook support

### PR DESCRIPTION
**Pull request by OliviaLynn**
Adding a rendered notebooks option to the template.

Highlights include:
* moving the `nb/` directory into 'docs/' (I found it the best way for the .rst referencing to remain intuitive--I don't think this loses too much over having it sit next to `src/`, but if anyone has a better method I'm all ears)
* converting `requirements.txt` into a jinja template to optionally include notebook dependencies 
* adding jinja whitespace specifiers to remove awkward spacing around conditionally-included text in toml, conf, etc.

Will be followed shortly with some writing in the RTD documentation on notebooks. I'd also like to add conditioning to the CI (left an `if: ${{ true }}` placeholder for now) to only install pandoc and other notebook requirements if option was selected, but this should be done in tandem with other CI changes I'm opening an issue for.

### CODE REVIEW

**Comment by drewoldag** [comment]
One slight change that I think is worth making here - I renamed the `nb` directory to `notebooks` in PR #83, so if you have a chance to make that change, that would be nice :) 

Also, I'm curious if you've had a chance to confirm that the pre-commit hook that will clear output from jupyter notebooks will pass the test immediately after hydrating a project from the template? I was running into an issue early on when I had a demo notebook where the pre-commit hook would find some small thing that had changed, and it would result in a failed pre-commit hook test for the very first commit.

**Review by drewoldag — APPROVED** [review]
This looks awesome. Thanks for putting it together!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -54,11 +54,17 @@ jobs:
       uses: psf/black@stable
       with:
         src: "../test/${{ matrix.copier_config.foldername }}/src"
+    - name: Install notebook requirements
+      if: ${{ true }}
```
Comment (delucchi-cmu) [inline .github/workflows/ci.yml:58]: Is this line necessary?

Comment (OliviaLynn) [reply .github/workflows/ci.yml:58]: It's the part I mention in the PR description above, where the `if: ${{ true }}` is a placeholder for conditioning this on whether or not the user has chosen the notebook option in the template. 

Strictly speaking it isn't necessary, but I plan to update it with actual logic as soon as I address issue #77.

**Comment by OliviaLynn** [comment]
> One slight change that I think is worth making here - I renamed the `nb` directory to `notebooks` in PR #83, so if you have a chance to make that change, that would be nice :)

Nice catch, thank you!

> Also, I'm curious if you've had a chance to confirm that the pre-commit hook that will clear output from jupyter notebooks will pass the test immediately after hydrating a project from the template? I was running into an issue early on when I had a demo notebook where the pre-commit hook would find some small thing that had changed, and it would result in a failed pre-commit hook test for the very first commit.

I haven't, but I'll admit I now default to editing notebooks in plaintext if possible with how messy things get in RAIL. I'll take a look once I get this in.

**Pull request merged**